### PR TITLE
Safeguard for secure store

### PR DESCRIPTION
**Pull request by gonzaloriestra**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/shopify-cli-planning/issues/261
Fixes https://github.com/Shopify/cli/issues/25

When saving session data to the secure store (through [keytar](https://www.npmjs.com/package/keytar)), it can fail in some cases:
- In Linux if libsecret is not installed (`libsecret-1.so.0: cannot open shared object file`)
- In Windows if the data is too long (`the stub received bad data`)
- In Mac if npm uses a wrong precompilation for keytar (`incompatible architecture`)

### WHAT is this pull request doing?

Use local storage if the secure store fails or if it's Windows, as a workaround until we reduce the size of our data. 

### How to test your changes?

`yarn dev` with at least one extension on Mac, Windows and Linux (with and without libsecret)

### CODE REVIEW

File: packages/cli-kit/src/secure-store.ts [hunk 1]
```diff
@@ -32,7 +50,38 @@ export async function store(identifier: string, content: string): Promise<void>
...
+  }
+}
+
+/**
+ * Returns true if the secure store is available on the system.
+ * Keytar it's not supported on some Linux environments or Windows.
+ * More details: https://github.com/Shopify/shopify-cli-planning/issues/261
+ * @returns a boolean indicating if the secure store is available.
+ */
+export async function secureStoreAvailable(
```
Comment (shauns) [inline packages/cli-kit/src/secure-store.ts:72]: If the behaviour we're going for is that users of `session/store.ts` use session storage transparently then I think we're better to not export this function, and have the switch in behaviour kept to just this module.

**Review by shauns — CHANGES_REQUESTED** [review]

File: packages/cli-kit/src/session/store.ts [hunk 2]
```diff
@@ -42,5 +58,32 @@ export async function fetch(): Promise<Session | undefined> {
...
+ * @returns a boolean indicating if the secure store is available.
+ */
+async function secureStoreAvailable(): Promise<boolean> {
+  try {
+    const keytar = await import('keytar')
+    await keytar.default.findCredentials(constants.keychain.service)
+    if (platformAndArch().platform === 'windows') {
+      debug(content`Secure store not supported on Windows`)
+      return false
+    }
```
Comment (pepicrft) [inline packages/cli-kit/src/session/store.ts:81]: What about moving to the top of the `try` and doing an early return?

File: packages/cli-kit/src/store.ts [hunk 3]
```diff
@@ -103,3 +104,18 @@ export function setThemeStore(store: string): void {
   debug(content`Setting theme store...`)
   cliKit.set('themeStore', store)
 }
+
+export function getSessionStore(): string | undefined {
```
Comment (pepicrft) [inline packages/cli-kit/src/store.ts:108]: Should we drop `store` from the name of the new functions in this module? (i.e. `getSession`, `setSession`, and `removeSession`). It's a bit misleading seeing `store` in the name when the module already represents a `store`.

Comment (gonzaloriestra) [reply packages/cli-kit/src/store.ts]: Agreed!

File: packages/cli-testing/src/temporary.ts [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
+/**
+ * Creates a temporary configuration store and ties its lifecycle to the callback.
+ * @param callback {(string) => void} Callback to execute. When the callback exits, the local config is destroyed.
+ * @returns {Promise<T>} Promise that resolves with the value returned by the callback.
+ */
+export function localConf<T>(callback: (localConf: store.LocalStore) => Promise<T>): Promise<T> {
+  const name = `shopify-cli-test-${uniqueString()}`
+  const localConfig = store.createConf(name)
+  // eslint-disable-next-line node/callback-return
+  const result = callback(localConfig)
```
Comment (pepicrft) [inline packages/cli-testing/src/temporary.ts:24]: You might want to use the following to ensure we also delete the directory in scenarios where the callback errors:
```js
try {

} finally {
  file.remove(localConfig.path)
}
```

**Pull request merged**